File: app/src/main/java/com/sparkstudios/taporiai/repository/TaporiRepository.kt
package com.sparkstudios.taporiai.repository

import com.sparkstudios.taporiai.network.ChatDownloadResponse
import com.sparkstudios.taporiai.network.ChatResponse
import com.sparkstudios.taporiai.network.CreditResponse
import retrofit2.Response

interface TaporiRepository {
    suspend fun sendMessage(
        idToken: String,
        chatId: String,
        prompt: String,
        systemMessage: String = "You are a Mumbai Tapori assistant. Reply in Mumbai slang hinglish language fully.",
        maxContextMessages: Int = 50
    ): Response<ChatResponse>

    suspend fun addCredit(idToken: String, creditsToAdd: Int): Response<CreditResponse>

    suspend fun downloadChat(idToken: String): Response<ChatDownloadResponse>
}

File: lib/features/chat/domain/repositories/tapori_repository.dart
import '../../../../core/error/result.dart';
import '../../../credits/domain/entities/credit_balance.dart';
import '../entities/chat_history.dart';
import '../entities/send_message_result.dart';

abstract interface class TaporiRepository {
  Future<Result<SendMessageResult>> sendMessage({
    required String idToken,
    required String chatId,
    required String prompt,
    String systemMessage =
        'You are a Mumbai Tapori assistant. Reply in Mumbai slang hinglish language fully.',
    int maxContextMessages = 50,
  });

  Future<Result<CreditBalance>> addCredit({
    required String idToken,
    required int creditsToAdd,
  });

  Future<Result<ChatHistory>> downloadChat({
    required String idToken,
  });
}
